backend/manager/modules/common/src/main/java/org/ovirt/engine/core/common/businessentities/comparators/VmsCpuPinningPolicyComparator.java

```diff
@@ -0,0 +1,28 @@
+package org.ovirt.engine.core.common.businessentities.comparators;
+
+import java.io.Serializable;
+import java.util.Comparator;
+
+import org.ovirt.engine.core.common.businessentities.VM;
+
+/**
+ * Comparing VMs based on the CPU pinning policy and if needed, their CPU topology.
+ */
+public class VmsCpuPinningPolicyComparator implements Comparator<VM>, Serializable {
+
+    @Override
+    public int compare(VM o1, VM o2) {
+        // Check the policy, their values are from the higher (more restricted policy) to the lower.
+        if (o2.getCpuPinningPolicy().getValue() - o1.getCpuPinningPolicy().getValue() != 0) {
```

Comment:
I think that if there already is a defined pinning (`MANUAL` or `RESIZE_AND_PIN`), it should be pinned first to the CPU topology, then `ISOLATED_THREADS` and finally `DEDICATED`.

Reply:
But the more restricted policies would be dedicated in terms of taking the physical resources while shared policies are not limiting anything but the exclusive policies.
I do think exclusive policy > shared pool policy in terms of restrictions.